python/tk_multi_loader/delegate_publish.py

```diff
@@ -108,16 +108,18 @@ class PublishDelegate(shotgun_view.EditSelectedWidgetDelegate):
     how things get rendered.
     """
 
-    def __init__(self, view, action_manager):
+    def __init__(self, view, action_manager, status_model=None):
```

Comment:
I don't think we should be passing this model around for use by external code like this hook. That's an internal, structural component of the app, and it should stay that way. It looks like you're wanting to get at the convenience method the model object provides to go from a status code to its long name, but that work can be done by the developer in the hook just using the normal API handle and a small amount of logic. I'd prefer that to be their responsibility.

Reply:
How can I access status name using the API inside hook?